src/System.Composition.Convention/tests/ConventionBuilderTests.cs

```diff
@@ -0,0 +1,202 @@
+                .SelectConstructor(cis => cis.ElementAtOrDefault(1));
+
+            var fooImplWithConstructors = typeof(FooImplWithConstructors).GetTypeInfo();
+
+            var constructor1 = fooImplWithConstructors.DeclaredConstructors.Where(c => c.GetParameters().Length == 0).Single();
+            var constructor2 = fooImplWithConstructors.DeclaredConstructors.Where(c => c.GetParameters().Length == 1).Single();
+            var constructor3 = fooImplWithConstructors.DeclaredConstructors.Where(c => c.GetParameters().Length == 2).Single();
+
+
+            // necessary as BuildConventionConstructorAttributes is only called for type level query for attributes
+            Assert.Equal(0, builder.GetCustomAttributes(typeof(FooImplWithConstructors), constructor1).Count());
+            Assert.Equal(0, builder.GetCustomAttributes(typeof(FooImplWithConstructors), constructor3).Count());
+
+            var ci = constructor2;
+            var attrs = builder.GetCustomAttributes(typeof(FooImplWithConstructors), ci);
+            Assert.Equal(1, attrs.Count());
+            Assert.Equal(typeof(ImportingConstructorAttribute), attrs.FirstOrDefault().GetType());
+        }
+
+        [Fact]
+        public void MapType_OverridingSelectionOfConventionSelectedConstructor_WithPartBuilderOfT()
+        {
+            var builder = new ConventionBuilder();
+
+            builder.
+                ForTypesDerivedFrom<IFoo>().
+                Export<IFoo>();
+
+            builder.ForType<FooImplWithConstructors>().
+                SelectConstructor(param => new FooImplWithConstructors(param.Import<IEnumerable<IFoo>>()));
+
+            var fooImplWithConstructors = typeof(FooImplWithConstructors).GetTypeInfo();
+
+            var constructor1 = fooImplWithConstructors.DeclaredConstructors.Where(c => c.GetParameters().Length == 0).Single();
+            var constructor2 = fooImplWithConstructors.DeclaredConstructors.Where(c => c.GetParameters().Length == 1).Single();
+            var constructor3 = fooImplWithConstructors.DeclaredConstructors.Where(c => c.GetParameters().Length == 2).Single();
+
+            // necessary as BuildConventionConstructorAttributes is only called for type level query for attributes
+            Assert.Equal(0, builder.GetCustomAttributes(typeof(FooImplWithConstructors), constructor1).Count());
+            Assert.Equal(0, builder.GetCustomAttributes(typeof(FooImplWithConstructors), constructor3).Count());
+
+            var ci = constructor2;
+            var attrs = builder.GetCustomAttributes(typeof(FooImplWithConstructors), ci);
+            Assert.Equal(1, attrs.Count());
+            Assert.Equal(typeof(ImportingConstructorAttribute), attrs.FirstOrDefault().GetType());
+        }
+
+        private interface IGenericInterface<T> { }
+
+        [Export(typeof(IGenericInterface<>))]
+        private class ClassExportingInterface<T> : IGenericInterface<T> { }
+        //[TestMethod]
```

Comment:
Should this commented out code be removed, or do we have plans to uncomment them (in which case an issue should be opened)?